bindings/matrix-sdk-ffi/src/event_filter.rs

```diff
@@ -0,0 +1,199 @@
+use ruma::events::{AnySyncTimelineEvent, TimelineEventType};
+
+use matrix_sdk_ui::timeline::event_type_filter::TimelineEventTypeFilter as InnerTimelineEventTypeFilter;
+
+#[derive(uniffi::Enum)]
+pub enum TimelineEventTypeFilter {
+    Include { event_types: Vec<FilterTimelineEventType> },
+    Exclude { event_types: Vec<FilterTimelineEventType> },
+}
+
+impl TimelineEventTypeFilter {
+    pub(crate) fn filter(&self, event: &AnySyncTimelineEvent) -> bool {
+        match self {
+            TimelineEventTypeFilter::Include { event_types } => {
+                let event_types: Vec<TimelineEventType> =
+                    event_types.iter().map(|t| t.clone().into()).collect();
+                InnerTimelineEventTypeFilter::Include(event_types).filter(event)
+            }
+            TimelineEventTypeFilter::Exclude { event_types } => {
+                let event_types: Vec<TimelineEventType> =
+                    event_types.iter().map(|t| t.clone().into()).collect();
+                InnerTimelineEventTypeFilter::Exclude(event_types).filter(event)
+            }
+        }
+    }
+}
+
+#[derive(uniffi::Enum, Clone)]
+pub enum FilterTimelineEventType {
+    MessageLike { event_type: FilterMessageLikeEventType },
+    State { event_type: FilterStateEventType },
+}
+
+impl From<FilterTimelineEventType> for TimelineEventType {
+    fn from(value: FilterTimelineEventType) -> TimelineEventType {
+        match value {
+            FilterTimelineEventType::MessageLike { event_type } => event_type.into(),
+            FilterTimelineEventType::State { event_type } => event_type.into(),
+        }
+    }
+}
+
+#[derive(uniffi::Enum, Clone)]
+pub enum FilterStateEventType {
+    PolicyRuleRoom,
+    PolicyRuleServer,
+    PolicyRuleUser,
+    RoomAliases,
+    RoomAvatar,
+    RoomCanonicalAlias,
+    RoomCreate,
+    RoomEncryption,
+    RoomGuestAccess,
+    RoomHistoryVisibility,
+    RoomJoinRules,
+    RoomMember,
+    RoomName,
+    RoomPinnedEvents,
+    RoomPowerLevels,
+    RoomServerAcl,
+    RoomThirdPartyInvite,
+    RoomTombstone,
+    RoomTopic,
+    SpaceChild,
+    SpaceParent,
+}
+
+#[uniffi::export]
+pub fn all_filter_state_event_types() -> Vec<FilterStateEventType> {
```

Comment:
These may be needed by iOS, since their exported enums don't conform to `CaseIterable` and don't have a `FilterStateEventType.allCases`. On Kotlin we do have `FilterStateEventType.entries` which contains this same info.

Reply:
I removed those methods in https://github.com/matrix-org/matrix-rust-sdk/pull/3025/commits/a46bd52b71e458d16ab9417167e5920c9ae6907b.